IO/BeagleBone/UARTFileBusBBB.cs

```diff
@@ -0,0 +1,152 @@
+                    case UARTRate.BAUD_4800: this.Port.BaudRate = 4800; break;
+                    case UARTRate.BAUD_9600: this.Port.BaudRate = 9600; break;
+                    case UARTRate.BAUD_19200: this.Port.BaudRate = 19200; break;
+                    case UARTRate.BAUD_38400: this.Port.BaudRate = 38400; break;
+                    case UARTRate.BAUD_57600: this.Port.BaudRate = 57600; break;
+                    case UARTRate.BAUD_115200: this.Port.BaudRate = 115200; break;
+                    case UARTRate.BAUD_230400: this.Port.BaudRate = 230400; break;
+                    case UARTRate.BAUD_460800: this.Port.BaudRate = 460800; break;
+                    default: this.Port.BaudRate = 0; break;
+                }
+            }
+        }
+        public UARTBitCount BitLength
+        {
+            get
+            {
+                switch (this.Port.DataBits)
+                {
+                    case 5: return UARTBitCount.BITS_5;
+                    case 6: return UARTBitCount.BITS_6;
+                    case 7: return UARTBitCount.BITS_7;
+                    case 8: return UARTBitCount.BITS_8;
+                }
+                return UARTBitCount.BITS_NONE;
+            }
+            set
+            {
+                switch (value)
+                {
+                    case UARTBitCount.BITS_5: this.Port.DataBits = 5; break;
+                    case UARTBitCount.BITS_6: this.Port.DataBits = 6; break;
+                    case UARTBitCount.BITS_7: this.Port.DataBits = 7; break;
+                    case UARTBitCount.BITS_8: this.Port.DataBits = 8; break;
+                    default: this.Port.DataBits = 0; break;
+                }
+            }
+        }
+        public UARTParity Parity
+        {
+            get
+            {
+                switch (this.Port.Parity)
+                {
+                    case System.IO.Ports.Parity.Even: return UARTParity.PARITY_EVEN;
+                    case System.IO.Ports.Parity.Odd: return UARTParity.PARITY_ODD;
+                }
+                return UARTParity.PARITY_NONE;
+            }
+            set
+            {
+                switch (value)
+                {
+                    case UARTParity.PARITY_EVEN: this.Port.Parity = System.IO.Ports.Parity.Even; break;
+                    case UARTParity.PARITY_ODD: this.Port.Parity = System.IO.Ports.Parity.Odd; break;
+                    default: this.Port.Parity = System.IO.Ports.Parity.None; break;
+                }
+            }
+        }
+        public UARTStopBits StopBits
+        {
+            get => ((this.Port.StopBits == System.IO.Ports.StopBits.One) ? UARTStopBits.STOPBITS_1 : UARTStopBits.STOPBITS_2);
+            set => this.Port.StopBits = ((value == UARTStopBits.STOPBITS_1) ? System.IO.Ports.StopBits.One : System.IO.Ports.StopBits.Two);
+        }
+        
+
+        public UARTFileBusBBB(string DevicePath)

```

Comment:
Add comment explaining how this communicates and for what it might be used. Having two types of UART busses on the BBB may get confusing otherwise.